tests/upgrades/test_errata.py

```diff
@@ -313,3 +317,255 @@ def test_post_scenario_errata_count_installtion(self):
         )
         for package in FAKE_9_YUM_UPDATED_PACKAGES:
             self._check_package_installed(client_container_id, package)
+
+
+class Scenario_errata_count_with_previous_version_katello_agent(APITestCase):
+    """The test class contains pre and post upgrade scenarios to test erratas count
+    and remotely install using n-1 'katello-agent' on content host.
+    Test Steps:
+        1. Before Satellite upgrade, Create a content host and register it with
+            Satellite
+        2. Install packages and down-grade them to generate errata.
+        3. Upgrade Satellite
+        4. Check if the Erratas Count in Satellite after the upgrade.
+        5. Install erratas remotely on content host and check the erratas count.
+    BZ: 1529682
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.docker_vm = settings.upgrade.docker_vm
+        cls.client_os = DISTRO_RHEL7
+        cls.org = entities.Organization().search(
+            query={'search': 'name="{}"'.format(DEFAULT_ORG)})[0]
+        cls.loc = entities.Location().search(
+            query={'search': 'name="{}"'.format(DEFAULT_LOC)})[0]
+
+    def _run_goferd(self, client_container_id):
```

Comment:
Scenario_errata_count, Scenario_errata_count_with_previous_version_katello_agent both class are using almost the same method, I think we can reuse it rather than creating a new one, what do you say?

Reply:
@devendra104 Yes I agree with it and noticed the same ... however this work require some refactor with exists tests as well hence I am thinking handle under separate issue/PR #7230 for them.

Reply:
:+1:

Reply:
ACK